### ISSUE TITLE
More series statuses

### ISSUE THREAD

**Issue by markus101** [Sonarr/Sonarr#1206]
Right now we have a few series statuses that we show in the UI.
- Continuing (Not cancelled, monitored)
- Ended (Cancelled)
- Paused (Continuing, not monitored)

There are some others might be useful as well/shown in different contexts.
- Mid-season (currently airing episodes daily/weekly)
- Mid-season break (Christmas break for example)
- End of season (to the best of our knowledge)
- Cancelled, but airing final episodes (IE. Perception was cancelled before the end of the final season, but still had episodes to air, in that case we showed it as ended, which it was due to be, but it was still on air).

**Comment by kayone** [comment]
I think we should separate the monitor status and series status, we combined them before for ui sake but that seems like a bad choice in hindsight.

**Comment by markus101** [comment]
I agree 100%, we're actually hiding the Ended and Unmonitored state too, so things can only get better by separating them.

**Comment by Taloth** [comment]
As long as we use reliable means to detect the hiatus status' instead of relying on metadata source status. But it can be tricky, some shows don't get their episodes added to tvdb until the episode airs, you don't want it to randomly detect a hiatus.

**Comment by Blizzke** [comment]
vaguely related but would be awesome nevertheless: A "seen" status per episode, so you wouldn't have to rely on other internet sites for that.

**Comment by markus101** [comment]
> As long as we use reliable means to detect the hiatus status' instead of relying on metadata source status.

It would need to be based on the episode information we have, but if we don't have future episodes and an episode recently aired (at least 2 weeks, maybe 3), then I wouldn't classify it as on hiatus to avoid that.

> vaguely related but would be awesome nevertheless: A "seen" status per episode, so you wouldn't have to rely on other internet sites for that.

I don't see the correlation, but something to consider if we ever start syncing watched statuses from other clients (Plex, Kodi, etc).

**Comment by MullinsN** [comment]
So at the moment, none of the TV series data sources hold the number of ordered episodes in a season?

If not, would be it be possible to request that this information is stored, and provided via the API?

**Comment by markus101** [comment]
Only by having the episodes listed, which Sonarr does know about, the problem is knowing if the episodes available is the complete list, there are often cases of shows not having episodes until the day/week they air so it could easily be unstable. Recently Grey's Anatomy only had 12 episodes listed for it's latest season until a day or two before the next episode was supposed to air, this was likely because the series as locked by TVDB admins, but is a prime example of the date not being accurate.

**Comment by MullinsN** [comment]
I appreciate that data can be out of date, given the sources, however in the majority of cases I would imagine that the initial season order is normally quite a solid number. I do appreciate that is never a piece of data I have spent a lot of time looking at :)

New series tend to be the most unsure, I would imagine.

So if I take an initial season 1 order of 16 episodes, and Sonarr gets to the point that 16 is reached, and looks ahead to find that E17 and E18 have appeared in the schedule for season 1, then the max number of episodes for this season would be 18 (At this time).  

Also if it was noted that the episode number had now exceeded max number of episodes originally thought to have been ordered, it would be an opportunity to request and update of series information, to see if the seasons max episodes had been updated. 

I appreciate there are other subtleties that I probably unaware of. 

I may never be totally accurate, but if the data was used more, say by Sonarr, then maybe the accuracy of the initial data would improve.
